packages/@lwc/engine-core/src/framework/hydration.ts

```diff
@@ -137,12 +234,29 @@ function hydrateNode(node: Node, vnode: VNode, renderer: RendererAPI): Node | nu
             hydratedNode = hydrateCustomElement(node, vnode, vnode.data.renderer ?? renderer);
             break;
     }
+
+    flushHydrationErrors(hydratedNode);
```

Comment:
Errors are queued as they occur and then logged with the source element once it has been hydrated and mounted to the DOM. Means the element in the console matches what is on the page and the highlighting works properly when you hover hover the elements in the console.